response_operations_ui/controllers/party_controller.py

```diff
@@ -31,6 +31,15 @@ def get_party_by_ru_ref(ru_ref):
     return response.json()
 
 
+def try_get_party_id_by_ru_ref(ru_ref):
```

Comment:
This function name isn't great.  But as well as that, why are you creating a whole new function for something that already exists?  There's a perfectly functional `get_party_by_ru_ref` function that does all your error handling for you.  You also use this function only once! You can use the other function and then extract the id or the empty string that way.
I get that you need to deal with the error case as you don't want it to go bang if it's not found.  I'll describe what to do in that bit

Reply:
Because get_party_by_ru_ref  raises api errors and I did not want that in this case. The user can enter any 11 digit string and they are not gaurenteed to be an ru ref.